### PR TITLE
Re-former app development

### PR DESCRIPTION
**Pull request by carlos-osorio-developer**
Add features for 

- Create new user
- Edit existing user
- Show errors messages
- Show user info

### CODE REVIEW

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Invalid Submission :x: :disappointed: :stop_sign: 
## Hello @carlos-osorio-developer and @pountzas   :handshake: :handshake: :handshake: 

Thank you for submitting a code review request. And congratulations on that you have completed the tic tac toe game. However your code review request is marked as invalid due to the following reasons:
- [x] Linters are failing
- [x] You are missing a descriptive readme file

Kindly implement those required basic requirements and submit for a review

Happy coding!!!
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Comment by carlos-osorio-developer** [comment]
> Thank you for submitting a code review request. And congratulations on that you have completed the tic tac toe game. 

You mean the Forms RoR project. Right?

**Comment by descholar-ceo** [comment]
> > Thank you for submitting a code review request. And congratulations on that you have completed the tic tac toe game.
> 
> You mean the Forms RoR project. Right?

Yes, sorry for this late reply, and please accept my apologies about this typo, I wanted to mention the forms ror project

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+
+    if @user.update(user_params)
+      redirect_to @user
```
Comment (descholar-ceo) [inline app/controllers/users_controller.rb]: - [x] [required] It should `redirect_to edit_user_path(<USER_ID>)` if user was saved (remember to change <USER_ID> to the code that works for the student implementation)

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :users, only: [:new, :create, :show, :update, :edit]
```
Comment (descholar-ceo) [inline config/routes.rb]: - [x] [required] `resources :users, only: [:new, :create, :edit, :update]` in config/routes.rb ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

=> Note that there is no show action

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<h1>Edit</h1>
+<%= render "form", user: @user %>
+
+
```
Comment (descholar-ceo) [inline app/views/users/new.html.erb]: - [x] You should have two commented out forms: 
  - [x] The first form uses native `html` tags
  - [x] The second uses `form_tag` rails helper method for generating html forms,
 => Kindly add them!

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @carlos-osorio-developer and @pountzas  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implement all of the requirements carefully, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] I have written all of the requirements remaining to take care of, check my inline comments in your codes and fi them accordingly!
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED :green_circle:
-----------
Hello @pountzas @carlos-osorio-developer  Great job working on the required changes :clap: :muscle: 
-------
- [x] [OPTIONAL] Remember that you must have 3 branches master branch + main development branch + feature branch. Please add the master branch. [Here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) is a reminder about how to work with GitFlow. Keep this in mind for the next projects. 
----------------
Now you can merge this branch :smile: :1st_place_medal: 
HAPPY CODING!!!! 

Well-Done-Applause

**Pull request merged**